subprojects/docs/src/docs/userguide/core-plugins/jacoco_plugin.adoc

```diff
@@ -30,7 +30,23 @@ include::sample[dir="snippets/testing/jacoco/quickstart/groovy",files="build.gra
 include::sample[dir="snippets/testing/jacoco/quickstart/kotlin",files="build.gradle.kts[tags=apply-plugin]"]
 ====
 
-If the Java plugin is also applied to your project, a new task named `jacocoTestReport` is created. Note that while tests should be executed before generation of report, `jacocoTestReport` task does not depend on the `test` task. By default, a HTML report is generated at `__$buildDir__/reports/jacoco/test`.
+If the Java plugin is also applied to your project, a new task named `jacocoTestReport` is created.
+By default, a HTML report is generated at `__$buildDir__/reports/jacoco/test`.
+
+[NOTE]
+====
+
+While tests should be executed before generation of the report, the `jacocoTestReport` task does not depend on the `test` task.
```

Comment:
Why don't we do this by default?

Reply:
Because it does trigger your tests to run when you generate the report (and the exec files may already be present even though your test task has run but is not up-to-date because it is not cachable). I agree that this should not be on the consumer side, we're working on a broader approach to this problem Just wanted to clarify the current situation.